go/logic/orchestrator.go

```diff
@@ -150,7 +155,10 @@ func discoverInstance(instanceKey inst.InstanceKey) {
 		return
 	}
 
-	if existsInCacheError := recentDiscoveryOperationKeys.Add(instanceKey.DisplayString(), true, cache.DefaultExpiration); existsInCacheError != nil {
+	// Calculate the expiry period each time as InstancePollSeconds
+	// _may_ change during the run of the process (via SIGHUP) and
+	// it is not possible to change the cache's default expiry..
+	if existsInCacheError := recentDiscoveryOperationKeys.Add(instanceKey.DisplayString(), true, instancePollSecondsDuration()); existsInCacheError != nil {
```

Comment:
May I suggest here using `instancePollSecondsDuration()*2` ? 
the real interval between two pools for an instance may be slightly greater than `instancePollSecondsDuration()` because life. It would be a pity to expire it before the next run.